portal/components/tokenSelector/list.tsx

```diff
@@ -0,0 +1,256 @@
+        <DatabaseIcon />
+        <h6 className="text-sm font-medium text-neutral-500">
+          {t('all-tokens')}
+        </h6>
+      </div>
+    </div>
+  )
+}
+
+const TokenRow = ({
+  offset,
+  onSelect,
+  start,
+  token,
+}: {
+  token: TokenType
+  start: number
+  offset: number
+  onSelect: (t: TokenType) => void
+}) => (
+  <li
+    className="absolute left-0 top-0 w-full cursor-pointer rounded-lg hover:bg-neutral-100"
+    onClick={() => onSelect(token)}
+    style={{
+      height: '56px',
+      // After showing your, all and custom tokens headers, we need to adjust the position of all the following tokens to consider it
+      transform: `translateY(${start + offset}px)`,
+    }}
+  >
+    <Token token={token} />
+  </li>
+)
+
+const shouldRenderHeader = (
+  alreadyRendered: boolean,
+  isFirst: boolean,
+  isSearchActive: boolean,
+  condition?: boolean,
+) => !alreadyRendered && isFirst && !isSearchActive && (condition ?? true)
+
+export const List = function ({
+  isSearchActive,
+  onSelectToken,
+  tokens,
+  yourTokens,
+}: Props & { isSearchActive: boolean } & {
+  yourTokens: TokenType[]
+}) {
+  const parentRef = useRef<HTMLDivElement>(null)
+  const { userTokenList } = useUserTokenList()
+
+  const combinedTokens = [...yourTokens, ...tokens, ...userTokenList]
+
+  const rowVirtualizer = useVirtualizer({
+    count: combinedTokens.length,
+    estimateSize: () => 56,
+    getScrollElement: () => parentRef.current,
+    overscan: 5,
+  })
+
+  const customTokensHeaderHeight = 40
+  const allTokensHeaderHeight = 60
+  const yourTokensHeaderHeight = 40
+
+  const getList = function () {
+    const rows: JSX.Element[] = []
+    const virtualItems = rowVirtualizer.getVirtualItems()
+    const hasYourTokens = yourTokens.length > 0
+
+    let hasAddedYourTokensHeader = false
+    let hasAddedAllTokensHeader = false
+    let hasAddedCustomTokensHeader = false
+
+    // Function to calculate the offset based on the headers that have been added
+    // This is used to adjust the position of the tokens after the headers
+    const getOffset = () =>
+      [
+        hasAddedYourTokensHeader && yourTokensHeaderHeight,
+        hasAddedAllTokensHeader && allTokensHeaderHeight,
+        hasAddedCustomTokensHeader && customTokensHeaderHeight,
+      ].reduce((acc, val) => acc + (val || 0), 0)
+
+    for (let rowIndex = 0; rowIndex < virtualItems.length; rowIndex++) {
+      const virtualItem = virtualItems[rowIndex]
+      const token = combinedTokens[virtualItem.index]
+
+      const isCustom = isCustomToken(userTokenList, token)
+      const isFirstYourTokens = rowIndex === 0
+      const isFirstAllTokens = virtualItem.index === yourTokens.length
+
+      const shouldRenderCustomHeader = shouldRenderHeader(
+        hasAddedCustomTokensHeader,
+        true, // custom tokens are always the last section
+        false, // custom tokens are not affected by search
+        isCustom,
+      )
```

Comment:
When there are more than 2 parameters, I prefer objects so it's easier to read and understand what each argument does without having to look to the signature of the method. Plus, you no longer depend on passing around the arguments in the appropriate order.


```suggestion
      const shouldRenderCustomHeader = shouldRenderHeader({
        alreadyRendered: hasAddedCustomTokensHeader,
        isFirst: true, // custom tokens are always the last section
        isSearchActive: false, // custom tokens are not affected by search
        condition isCustom,
      })
```



What do you think?